### DISCUSSION TITLE
MissionControl 0.8.0 prerelease

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#521] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.8.0/total">
</p>

This is the sixteenth official release of Mission Control.

This is purely a compatibility release to support the newly released 15.0.0 firmware, since I pushed all my recent changes out with the 0.7.1 release last week.

Please note that this is a pre-release intended to allow people to continue using the module on 15.0.0 while I finalise support for it. **It should be stable** (though please report any issues if you encounter them), but is missing the patches to the `bluetooth` module for 15.0.0 that enable new Wii and Xbox Elite 2 controllers to be paired.

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl#readme) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

### Changelog:
- Updated libstratosphere for 15.0.0 support

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.8.0'>MissionControl 0.8.0 prerelease</a>.</em>

**Comment by androgit-ops** [comment]
you already addressed the complaint i  was about to drop,cant pair new wii u pro bt controller

**Reply by ndeadly** [reply]
Yep, facing some challenges with the bluetooth patches due to having very little unused contiguous code memory available to store them on this version. I'm working on it, just haven't had a lot of free time this past week.

**Reply by androgit-ops** [reply]
thanks a lot for putting effort in this

**Comment by androgit-ops** [comment]
I just gave up 5 minutes ago after 2 days try, i use the switch in a games
lounge

**Comment by Hoenhaimen** [comment]
Thank you very much for your effort, I swear if I knew how to program it would help you with this work, but tell me there would be any predictions to add the patch for wiiu control, I also need it.

**Reply by ndeadly** [reply]
Hopefully over the weekend if I can find the time.

**Reply by Hoenhaimen** [reply]
tks man! You are the best.

**Comment by SafeNut** [comment]
any chance that support for Xbox Series controllers will be coming soon?

**Reply by Masamune3210** [reply]
They require Bluetooth LE support, which has yet to be worked out yet. Trust me, there will be lots of fanfare when/if it is

**Reply by ndeadly** [reply]
It's my number one priority after finishing the patches for 15.0.0. I have it working as a proof of concept but there are several issues that make it not suitable for release. To date, trial and error hasn't worked in figuring them out, so I am resorting to extensively reverse engineering the bluetooth and btm sysmodules to try and learn how certain things are handled in official software. It's hard to say how long this may take due to the uncertain nature of figuring out the unknowns.

**Reply by AlexSpaces** [reply]
so not yet huh
just as expected

**Reply by ndeadly** [reply]
Feel free to implement support yourself.

**Reply by 4bosch** [reply]
Thank you for everything you do @ndeadly and don't let stupid people demoralize ! :D

**Reply by AlexSpaces** [reply]
:(

**Comment by jacobo2110** [comment]
Good morning, thank you very much for your contributions to the community, I wrote to ask you if you could make the saitake 7007f control compatible in the next version, I would really appreciate it.

**Reply by ndeadly** [reply]
I am not a wizard, and cannot add support for controllers I know nothing about. Please follow the instructions to create a [controller request](https://github.com/ndeadly/MissionControl/issues/new/choose) and fill out the required info.

**Reply by jacobo2110** [reply]
I  just finished the controller request to add the Nintendo Switch controller support for this Saitake STK-7007F model. Thanks in advance for the quick response.

**Comment by valenparra** [comment]
Hello. Thanks a lot for your work! The existence of MissionControl itself justifies going through all the Switch unlocking procedure.
I've tested with WiiU Pro controller, Dualshock4 V2 and Wiimote (wii motion inside version). All of them detected right away and working fine. Wireless motion also works but has some issues: with the Wiimote, left-right movement is inversed with up-down movements, and with Dualshock it works almost perfect, but the movements are somehow tilted (a horizontal left-right movement is not being read as horizontal, but a bit tilted counter clockwise. Is there a way to calibrate wireless motion? Thanks again and keep it up with your awesome work!

**Reply by ndeadly** [reply]
The orientation of the wiimote motion axis is intentional. Without a nunchuck or other accessory attached, the wiimote is assumed to be held in the horizontal orientation. It will automatically switch to vertical orientation when an accessory is detected. There are some issues with reliably detecting accessories when Motionplus is involved, so it's also possible you are using a nunchuck etc but it's not being detected correctly. This can usually be resolved by disconnecting and reconnecting the attachment a few times.

As for the ps4 controller, the incoming data is already calibrated with the calibration data fetched from the controller. From memory I think the console has an option to perform a user calibration somewhere in the system settings but I'm not sure how effective it would be on something like that.   Does the controller only exhibit the rotated axes using mission control?

**Reply by valenparra** [reply]
Thanks a lot for your quick answer. After reading it, I have tested again (I test on Super Mario Galaxy from Super Mario All Stars):

- Connecting and disconnecting the nunchuck on the Wiimote is well recognized, because the nunchuck's joystick and other buttons work, and the "re-center pointer / cursor" button changes (from "A" button on wiimote-only mode, to "1" button on wiimote+nunchuck mode). But the movements are still inversed. I have tried re-pairing the wiimote on the console's Controllers->Change Grip/Order menu with the nunchuck connected in the moment of pairing, but same result. I also tried disconnecting and connecting again the nunchuck, but no chance with that, motion controls are still inversed

- I have also tried with Dualshock 4 controller. I was not able to check with PlayStation 4 calibration, cause I don't have one, I just have the Dualshock 4 controller for gaming with my Android phone, and now for my Switch :). But I got to an interesting conclusion: when connecting the controller through Controllers->Change Grip/Order, if the Dualshock 4 controller is horizontal, when I get back to the game, the axes are not rotated any more. So I tried connecting again the controller on Controllers->Change Grip/Order screen, with the controller in a tilted position, and there you go => rotated axes! So: wireless motion seems to be taking the axis references based on the controller position in the moment of connecting it to the console. The Dualshock 4 controller should always be placed on an horizontal surface when connecting it to the console to get correct axis positions when using wireless motion in game.

I hope this testing information is useful for other Dualshock 4 users, and that maybe a fix can be found for Wiimote wireless motion axis modes with / without nunchuck. Playing Super Mario Galaxy with a Wiimote and nunchuck is just so much better gaming experience, I still play it on my Wii sometimes, but visuals are better on Switch version

**Reply by ndeadly** [reply]
That's interesting, I didn't know that about the Dualshock controller. On that topic though, what's the orientation of your wiimote when you connect it? You need to have it sitting button side down on a flat surface when you connect it or else the motion controls will also be off. Maybe that's the cause of your problem? I don't own a motion plus controller myself, but I had somebody verify the motion axes and supposedly everything is ok with their controller. Also, you don't need to go through the Change Grip/Order screen for anything after the controller has been paired the first time. It's enough to just hit one of the face buttons to wake it up and it will automatically try to connect.

**Comment by impeeza** [comment]
@ndeadly Dude, you are the best, thanks a lot for all your work, I know is hard and the rough requests do it harder. please receive the kindest regards from Colombia.

**Comment by CoseZZZ** [comment]
@ndeadly Is it compatible with 15.0.1 because when i try to start it with hekate it says:
A fatal error ocurred when running Fusee.
Failed to open sdmc:/atmosphere/package3!
Press POWER to reboot.

Any solutions?

**Reply by ndeadly** [reply]
Mission Control is compatible with 15.0.1, yes. This sounds like a problem with either your atmosphere installation or the sd card itself. You'll need to seek help for that elsewhere.

**Reply by CoseZZZ** [reply]
Thank you.

**Comment by alamalamala** [comment]
Hello, I know it's a lot to ask, but could you fix the problem with the smash? what happens when you select insert amibo, it slows down the entire system and a hard reset must be done this happens in smash bros ultimate and thanks for your great effort

**Comment by Juanmikun** [comment]
Hi, thanks a lot for your work, it's been amszing to be able to give a new life to old remotes!
Since updating switch os to 15 and latest update of missioncontrol I have a problem wirh the Wii U remote pro, wirh the left joystick. It used to work well, but now sometimes left joystick won't respond well, specially when pressing up. This happens while playing pikmin 3 deluxe. Does anyone else have problems with wii u remote Pro? As i said, it was working flawlessly until latest update.

Thank you very much again for this amazing work!

**Reply by ndeadly** [reply]
This sounds like a stick calibration issue. Wii U controllers don't have an onboard factory calibration like switch controllers do, and should be calibrated using the option in the system menu. My guess would be either you never did this and just didn't notice the issues before, or you've changed sd cards or deleted `/config/MissionControl/controllers` where the calibration gets stored.

**Reply by Juanmikun** [reply]
Hi, thanks a lot for replying :) 
I actually did the calibration twice yesterday before posting here and unfortunately it doesn't solve the issue. In fact, left stick wiggles when I move it and never reach the end. This was not happening before. Right stick works fine. There is no hardware problem because it works fine on wii u.
I didn't delete that folder, and, in fact, trying go fix it, reinstalled again from the homebrew store... So I don't know what could be. I also tried to re-pair the remote... Running out of ideas 😂

**Comment by bffb42** [comment]
Looking forward to supporting the latest Xbox series

**Comment by martinmagic** [comment]
Need to support Luna Amazon controller Bluetooth pairing thanks

**Reply by ndeadly** [reply]
Please open a controller request issue. There's more to supporting it than simply knowing it's name.
